File: atcoder/abc337a_scoreboard.cpp
// Vicfred
// https://atcoder.jp/contests/abc337/tasks/abc337_a
// implementation
#include <iostream>

using namespace std;

int main() {
  int N;
  cin >> N;
  int X, Y;
  X = Y = 0;
  for(int i = 0; i < N; ++i) {
    int x, y;
    cin >> x >> y;
    X += x;
    Y += y;
  }
  if (X < Y) {
    cout << "Aoki" << endl;
  } else if (X > Y) {
    cout << "Takahashi" << endl;
  } else {
    cout << "Draw" << endl;
  }
  return 0;
}

File: atcoder/abc337a_scoreboard.hs
import Data.List (transpose)
import Control.Monad (replicateM)
-- vicfred
-- https://atcoder.jp/contests/abc337/tasks/abc337_a
-- implementation
main :: IO ()
main = do
  n <- read <$> getLine :: IO Int
  [takahashi, aoki] <- transpose <$> replicateM n (map read . words <$> getLine) :: IO [[Int]]
  putStrLn $ whoWins takahashi aoki

whoWins :: [Int] -> [Int] -> String
whoWins x y
        | takahashi > aoki = "Takahashi"
        | takahashi < aoki = "Aoki"
        | otherwise = "Draw"
        where takahashi = sum x
              aoki      = sum y
